lib/ansible/playbook/base.py

```diff
@@ -165,7 +165,7 @@ class Base(with_metaclass(BaseMeta, object)):
     _ignore_errors       = FieldAttribute(isa='bool')
     _check_mode          = FieldAttribute(isa='bool')
     _diff                = FieldAttribute(isa='bool')
-    _any_errors_fatal     = FieldAttribute(isa='bool', default=C.ANY_ERRORS_FATAL, always_post_validate=True)
+    _any_errors_fatal     = FieldAttribute(isa='bool', always_post_validate=True)
```

Comment:
so this change i think is the base of the fix, but instead of what you are doing below, just make sure you override the `post validate` for the `play` object's `any_errors_fatal` and assign the default there if `None`. The rest should fall into place via inheritance.

Reply:
The other option is to remove here but post validate at task level after checking inhertiance, set to the default if `None`, removing also `always_post_validate`